rust/core-lib/src/editor.rs

```diff
@@ -998,6 +998,17 @@ impl<W: Write + Send + 'static> Editor<W> {
     // Note: the following are placeholders for prototyping, and are not intended to
     // deal with asynchrony or be efficient.
 
+    /// Applies an async edit from a plugin.
+    pub fn plugin_edit(&mut self, edit: &PluginEdit) {
+        self.this_edit_type = EditType::Other;
+        let interval = Interval::new_closed_open(edit.start as usize, edit.end as usize);
+        let text = Rope::from(&edit.text);
+        let rev_len = self.engine.get_rev(edit.rev).unwrap().len();
+        let delta = Delta::simple_edit(interval, text, rev_len);
```

Comment:
We definitely eventually want to support full multi-edit Deltas from plugins because that is our story for how to cleanly handle asynchronous formatting and file reloading.

I would say add a TODO here and soon before building up too much stuff based on edits from plugins I would appreciate a refactor of the plugin API to send a full Delta representation. Perhaps even just based on the serde implementations for `Delta`.